lib/ansible/modules/network/enos/enos_facts.py

```diff
@@ -0,0 +1,497 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+#
+# Copyright (C) 2017 Lenovo, Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Module to Collect facts from Lenovo Switches running Lenovo ENOS commands
+# Lenovo Networking
+#
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: enos_facts
+version_added: "2.5"
+author: "Anil Kumar Muraleedharan (@amuraleedhar)"
+short_description: Collect facts from remote devices running Lenovo ENOS
+description:
+  - Collects a base set of device facts from a remote Lenovo device
+    running on ENOS.  This module prepends all of the
+    base network fact keys with C(ansible_net_<fact>).  The facts
+    module will always collect a base set of facts from the device
+    and can enable or disable collection of additional facts.
+extends_documentation_fragment: enos
+notes:
+  - Tested against ENOS 8.4.1.68
+options:
+  gather_subset:
+    description:
+      - When supplied, this argument will restrict the facts collected
+        to a given subset.  Possible values for this argument include
+        all, hardware, config, and interfaces.  Can specify a list of
+        values to include a larger subset.  Values can also be used
+        with an initial C(M(!)) to specify that a specific subset should
+        not be collected.
+    required: false
+    default: '!config'
+'''
+EXAMPLES = '''
+Tasks: The following are examples of using the module enos_facts.
+---
+- name: Test Enos Facts
+  enos_facts:
+    provider={{ cli }}
+
+  vars:
+    cli:
+      host: "{{ inventory_hostname }}"
+      port: 22
+      username: admin
+      password: admin
+      transport: cli
```

Comment:
`transport` option is not present in `provider` spec in `module_uitls/enos.py`. Is this required?

Reply:
Removed.